File: heranca_multipla.py
# noinspection PyMethodMayBeStatic
class Terrestre():

    def __init__(self, quant_rodas):
        self.rodas = quant_rodas

    def caminhar(self):
        print("Caminhando...")

    def parar(self):
        print("Parando de caminhar")


class Aquatico():

    def __init__(self, quant_propulsores):
        self.propulsores = quant_propulsores

    def nadar(self):
        print("Nadando...")

    def parar(self):
        print("Parando de nadar")


class Hibrido(Terrestre, Aquatico):

    def __init__(self, q_rodas, q_propulsores):
        Terrestre.__init__(self, q_rodas)
        Aquatico.__init__(self, q_propulsores)

    def migrar(self):
        print("Migrando de ambiente...")

    def parar(self):
        print("Parando de mover")


def main():
    robo_hibrido = Hibrido(4, 6)
    print(f"Rodas: {robo_hibrido.rodas}")
    print(f"Propulsores: {robo_hibrido.propulsores}")
    Terrestre.parar(robo_hibrido)
    Aquatico.parar(robo_hibrido)


if __name__ == '__main__':
    main()

File: heranca_multipla.cpp
#include <iostream>

class Terrestre {
public:
    int rodas;

    explicit Terrestre(int quant_rodas) : rodas(quant_rodas) {}

    void caminhar() {
        std::cout << "Caminhando..." << std::endl;
    }

    void parar() {
        std::cout << "Parando de caminhar" << std::endl;
    }
};

class Aquatico {
public:
    int propulsores;

    explicit Aquatico(int quant_propulsores) : propulsores(quant_propulsores) {}

    void nadar() {
        std::cout << "Nadando..." << std::endl;
    }

    void parar() {
        std::cout << "Parando de nadar" << std::endl;
    }
};

class Hibrido : public Terrestre, public Aquatico {
public:
    Hibrido(int q_rodas, int q_propulsores)
            : Terrestre(q_rodas), Aquatico(q_propulsores) {}
};

int main() {
    Hibrido robo_hibrido(4, 6);
    std::cout << "Rodas: " << robo_hibrido.rodas << std::endl;
    std::cout << "Propulsores: " << robo_hibrido.propulsores << std::endl;
    robo_hibrido.Terrestre::parar();
    robo_hibrido.Aquatico::parar();
    return 0;
}

